humilis/layer.py

```diff
@@ -280,47 +283,58 @@ def create(self, update=False):
                 self.name, self.cf_name))
 
             cf_template = json.dumps(self.compile(), indent=4)
-            try:
-                self.cf.create_stack(
-                    self.cf_name,
-                    cf_template,
-                    self.sns_topic_arn,
-                    self.tags)
-                self.wait_for_status_change()
-            except ClientError:
-                self.logger.error(
-                    "Error deploying stack '{}'".format(self.cf_name))
-                self.logger.error("Stack template: {}".format(cf_template))
-                raise
+            self.create_with_changeset(cf_template)
         elif update:
-            self.logger.info("Updating layer '{}'".format(self.name))
             cf_template = json.dumps(self.compile(), indent=4)
             try:
-                self.cf.update_stack(
-                    self.cf_name,
-                    cf_template,
-                    self.sns_topic_arn)
-                self.wait_for_status_change()
+                self.create_with_changeset(cf_template, update)
             except NoUpdatesError:
-                msg = "No updates on layer '{}'".format(self.name)
+                msg = "Nothing to update on stack '{}'".format(self.cf_name)
                 self.logger.warning(msg)
-            except:
-                self.logger.error(
-                    "Error deploying stack '{}'".format(self.cf_name))
-                self.logger.error("Stack template: {}".format(cf_template))
-                raise
         else:
             msg = "Layer '{}' already in CF: not creating".format(self.name)
             self.logger.info(msg)
 
-
         return self.outputs
 
+    def create_with_changeset(self, cf_template, update=False):
```

Comment:
All the code above has been merged in this method.
The behaviour is not the same depending if it is an update or a create.
For the update we need to check the `["Changes"]` list of the created change_set to see if there really is something to update on the stack.